dotCMS/src/integration-test/java/com/dotcms/datagen/AppDescriptorDataGen.java

```diff
@@ -0,0 +1,237 @@
+package com.dotcms.datagen;
+
+import com.dotcms.security.apps.AppDescriptor;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.security.apps.ParamDescriptor;
+import com.dotcms.security.apps.Type;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.collect.Ordering;
+import com.liferay.util.StringPool;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.SortedMap;
+
+/**
+ * Test utility to simplify creation of application descriptor yml files
+ */
+public class AppDescriptorDataGen extends AbstractDataGen<AppDescriptor> {
+
+    private static final ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory())
+            .findAndRegisterModules();
+
+    private String key = String.format("app_%d", System.currentTimeMillis());
+    private String name = key;
+    private String fileName = String.format("%s.yml", key);
+    private  String description = "";
+    private Boolean allowExtraParameters;
+    private String iconUrl = "none.ico";
+
+    private SortedMap<String, ParamDescriptor> paramMap;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Field paramMap has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)